src/components/ElementSearchDialog/element-search-input.tsx

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import {
+    Autocomplete,
+    AutocompleteProps,
+    AutocompleteRenderInputParams,
+} from '@mui/material';
+import { HTMLAttributes, ReactNode, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+
+export type RenderElementProps<T> = HTMLAttributes<HTMLLIElement> & {
+    element: T;
+    inputValue: string;
+    onClose?: () => void;
+};
+
+export interface ElementSearchInputProps<T>
+    extends Pick<
+        AutocompleteProps<T, false, false, true>,
+        'sx' | 'size' | 'loadingText' | 'loading'
+    > {
+    searchTerm: string;
+    onClose?: () => void;
+    onSearchTermChange: (searchTerm: string) => void;
+    onSelectionChange: (selection: T) => void;
+    getOptionLabel: (option: T) => string;
+    isOptionEqualToValue: (option1: T, option2: T) => boolean;
+    elementsFound: T[];
+    renderElement: (props: RenderElementProps<T>) => ReactNode;
+    renderInput: (
+        searchTerm: string,
+        props: AutocompleteRenderInputParams
+    ) => ReactNode;
+    searchTermDisabled?: boolean;
+    searchTermDisableReason?: string;
+    showResults?: boolean;
+}
+
+export const ElementSearchInput = <T,>(props: ElementSearchInputProps<T>) => {
+    const {
+        elementsFound,
+        loading,
+        onSearchTermChange,
+        onSelectionChange,
+        renderElement,
+        renderInput,
+        getOptionLabel,
+        isOptionEqualToValue,
+        onClose: handleClose,
+        showResults,
+        searchTerm,
+        loadingText,
+        searchTermDisableReason,
+        searchTermDisabled,
+        size,
+        sx,
+    } = props;
+
+    const intl = useIntl();
+
+    const displayedValue = useMemo(() => {
+        return searchTermDisabled || searchTermDisableReason
+            ? searchTermDisableReason ??
+                  intl.formatMessage({
+                      id: 'element_search/searchDisabled',
+                  })
+            : searchTerm ?? '';
+    }, [searchTerm, searchTermDisabled, searchTermDisableReason, intl]);
+
+    return (
+        <Autocomplete
+            sx={sx}
+            open={showResults}
+            freeSolo
+            size={size}
+            id="element-search"
+            forcePopupIcon={false}
+            fullWidth
+            onInputChange={(_event, value, reason) => {
+                if (!searchTermDisabled && (reason !== 'reset' || !value)) {
+                    onSearchTermChange(value);
+                }
```

Comment:
The logic here is confusing, why is there `|| !value` here ?

Reply:
it was here because of a something I've removed and is now obsolete
removed